ctapipe_io_magic/laser_params_corr_test.py

```diff
@@ -0,0 +1,411 @@
+                    value = value.split("#")[0].strip()  # Remove comments
+                if variable == "gkIntegrationWindow":  # Fix indentation here
+                    try:
+                        gkIntegrationWindow = float(value)
+                    except ValueError:
+                        raise ValueError("Invalid value for gkIntegrationWindow in params.txt")
+
+        if gkIntegrationWindow is None:
+            raise ValueError("Variable gkIntegrationWindow not found in params.txt")
+
+        signalsamples = 0
+        sig = np.zeros(signalsamples)
+        if transmission6km < -998.:
+            for i in range(466, signalsamples):
+                background1 += np.power((phecounts[i] - phecounts[i - 1]) * 1E6 / np.power(gkIntegrationWindow * i, 2), 2)
+            background1 /= 30.0
+            bg = background1
+            for i in range(signalsamples):
+                sig[i] = pheounts[i] * 1E6 / np.power(gkIntegrationWindow * i, 2) + background1
+        return sig
+
+    def get_range(self, i, bgsamples, ncollapse, t0shift):
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/params.txt'
+        gkBGSamplesV1to4 = None
+        gkIntegrationWindow = None
+        with open(filename, 'r') as file:
+            for line in file:
+                exec(line.strip())
+                if line.strip():  # Check if line is not empty
+                    variable, value = line.split("=")
+                    variable = variable.strip()
+                    value = value.split("#")[0].strip()
+                if variable == "gkBGSamplesV1to4":
+                    try:
+                        gkBGSamplesV1to4 = int(value)
+                    except ValueError:
+                        raise ValueError("Invalid value for gkBGSamplesV1to4 in params.txt")
+                elif variable == "gkIntegrationWindow":  # Fix indentation here
+                    try:
+                        gkIntegrationWindow = float(value)
+                    except ValueError:
+                        raise ValueError("Invalid value for gkIntegrationWindow in params.txt")
+
+        if gkBGSamplesV1to4 is None:
+            raise ValueError("Variable gkBGSamplesV1to4 not found in params.txt")
+
+        if gkIntegrationWindow is None:
+            raise ValueError("Variable gkIntegrationWindow not found in params.txt")
+
+        ifadc = i + bgsamples
+        return (self.get_offset_r(ifadc, bgsamples, ncollapse) + gkIntegrationWindow * (self.get_fadcr(i, bgsamples) + 0.5) * self.get_bin_width(ifadc, bgsamples, ncollapse)
+            + t0shift * gkIntegrationWindow * self.get_bin_width(ifadc, bgsamples, ncollapse))
+
+    def get_signal(self, i, bgsamples, phecounts, ncollapse, bg, hwcorr):
+        ifadc = i + bgsamples
+        return ((phecounts - self.get_bin_width(ifadc, bgsamples, ncollapse) * bg) * hwcorr
+            / self.get_bin_width(ifadc, bgsamples, ncollapse))
+
+def main():
```

Comment:
this is fine for tests, but ultimately you should just execute the relevant functions from __init__.py file and store the calibrated LIDAR information in the data structure (like you do with uncalibrated)